### PR TITLE
Abstract key storage and signing operations to facilitate different key storage types

### PR DESCRIPTION
**Pull request by PManaras**
This PR introduces support for implementing different types of keystorages via the newly KeyProvider interface. This interface can be used to retrieve keys from different types of storages, i.e. remote vaults, file-system, database.

The newly created SigningService can be used to implement different types of signature processes. The standard signature process is "LOCAL" meaning the MIW will use keys provided by the keyprovider to sign VP/VC locally.

This PR also adds to new properties to application.yaml (authoritySigningServiceType (defaults to "LOCAL), localSigningKeyStorageType (defaults to DB). These additions are also reflected in the corresponding env-files.

The database entity Wallet was updated and now contains a new field signingServiceType that reflects which wallet uses which signatureServices. The new field is introduced via a new database migration (v3) and sets the signing_service_type to "LOCAL" for all existing wallets. This means: with the standard configuration the MIW will act as before.

this PR includes changes made in the [PR on the main-branch](https://github.com/eclipse-tractusx/managed-identity-wallet/pull/276)

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

This PR is qual to [PR285](https://github.com/eclipse-tractusx/managed-identity-wallet/pull/285) from a different merge source

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
 # [0.5.0-develop.13](https://github.com/eclipse-tractusx/managed-identity-wallet/compare/v0.5.0-develop.12...v0.5.0-develop.13) (2024-04-18)
 
-
```
Comment (borisrizov-zf) [inline CHANGELOG.md:3]: Re-add this line, as it is present in our version of the develop branch.

File: src/main/java/org/eclipse/tractusx/managedidentitywallets/ManagedIdentityWalletsApplication.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /*
  * *******************************************************************************
- *  Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/managedidentitywallets/ManagedIdentityWalletsApplication.java:3]: Revert to 2023 in this file, as nothing else has changed.

The date is only changed in modified files.

**Review by borisrizov-zf — COMMENTED** [review]
Two minor changes needed due to divergence in source branches, after that we will wait for the IP check approval.

**Comment by nitin-vavdiya** [comment]
License headers need to be updated in the changed file

**Comment by borisrizov-zf** [comment]
@thackerronak @nitin-vavdiya We're ok to rebase this branch and do final testing before we merge.

**Comment by nitin-vavdiya** [comment]
> @thackerronak @nitin-vavdiya We're ok to rebase this branch and do final testing before we merge.

Rebase is done, fixing failing test cases, will let  you know once it is ready to merge

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]

File: src/main/java/org/eclipse/tractusx/managedidentitywallets/dao/repository/WalletKeyRepository.java [hunk 3]
```diff
@@ -33,20 +33,43 @@ public interface WalletKeyRepository extends BaseRepository<WalletKey, Long> {
     /**
      * Gets by wallet id and algorithm.
      *
-     * @param id the id
-     * param algorithm the algorithm
+     * @param id        the id param algorithm the algorithm
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply src/main/java/org/eclipse/tractusx/managedidentitywallets/dao/repository/WalletKeyRepository.java]: fixed with: https://github.com/eclipse-tractusx/managed-identity-wallet/pull/291/commits/fcf86af9177afa6d516b34367a45d6ab02214eaf

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,82 @@
 
 ### Bug Fixes
 
+* add asJwt as query param and fix
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply CHANGELOG.md:16]: This is merge commit I think after rebase

**Pull request merged**